pvanalytics/system.py

```diff
@@ -401,3 +403,177 @@ def infer_orientation_daily_peak(power_or_poa, sunny, tilts,
                 best_azimuth = azimuth
                 best_tilt = tilt
     return best_azimuth, best_tilt
+
+
+def _power_residuals_from_clearsky(system_params,
+                                   ghi, dhi, dni,
+                                   solar_zenith, solar_azimuth,
+                                   power_ac, temperature,
+                                   wind_speed,
+                                   temperature_coefficient,
+                                   temperature_model_parameters):
+    """Return the residuals between a system with parameters given in `params`
+    and the data in `power_ac`.
+
+    Parameters
+    ----------
+    system_params : array-like
+        array of four floats: tilt, azimuth, DC capacity, and inverter
+        DC input limit.
+    ghi : Series
+        Clear sky GHI
+    dhi : Series
+        Clear sky DHI
+    dni : Series
+        Clear sky DNI
+    solar_zenith : Series
+        Solar zenith at the same times as data in `power_ac`
+    solar_azimuth : Series
+        Solar azimuth at the same times as data in `power_ac`
+    power_ac : Series
+        Measured AC power under clear sky conditions.
+    temperature : float or Series
+        Air temperature at which to model the hypothetical system. If a
+        float then a constant temperature is used. If a Series, must have
+        the same index as `power_ac`. [C]
+    wind_speed : float or Series
+        Wind speed. If a float then a constant wind speed is used. If a
+        Series, must have the same index as `power_ac`. [m/s]
+    temperature_model_parameters : dict
+        Parameters fot the cell temperature model.
+
+    Returns
+    -------
+    Series
+        Difference between `power_ac` and the PVWatts output with the
+        given parameters.
+    """
```

Comment:
```suggestion
    
    Notes
    ------
    Uses the defaults in `pvlib.irradiance.get_total_irradiance` to calculated plane-of-array
    irradiance, i.e., the isotropic model for sky diffuse irradiance, and the Perez irradiance
    transposition model.
    """
```

Reply:
@cwhanse I don't think this code is actually using Perez.  Is that something we should update?  Seems like a shame to deviate from PVWatts in a way that will presumably make the fits worse.

cc @kperrynrel since she's getting bad fits :)

Reply:
Alternatively, should we switch the code to use Perez?  I think the extra inputs could be calculated automatically.  It would be more aligned with PVWatts itself, plus hopefully return better results.

Reply:
Fair point.  Note that `ModelChain.with_pvwatts` currently uses the Sandia temperature model too: https://github.com/pvlib/pvlib-python/blob/master/pvlib/modelchain.py#L49-L59

I'll open an issue.

Reply:
#147